packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js

```diff
@@ -82,38 +80,45 @@ export default ({ coreSagas }) => {
       yield put(A.coinifyLoading())
       const trade = yield call(coreSagas.data.coinify.sell)
 
+      const state = yield select()
+
       if (!trade) {
         const trade = yield select(selectors.core.data.coinify.getTrade)
         const parsed = JSON.parse(trade.error)
 
         yield put(A.coinifyFailure(parsed))
         return
       }
-      const p = yield select(sendBtcSelectors.getPayment)
+      const p = path(['coinify', 'payment'], state)
       let payment = yield coreSagas.payment.btc.create(
         { payment: p.getOrElse({}), network: settings.NETWORK })
       payment = yield payment.amount(parseInt(trade.sendAmount))
 
       // QA Tool: manually set a "to" address on the payment object for testing sell
-      const qaState = yield select()
-      const qaAddress = path(['qa', 'qaSellAddress'], qaState)
-
+      const qaAddress = path(['qa', 'qaSellAddress'], state)
       if (qaAddress) {
         payment = yield payment.to(qaAddress)
       } else {
         payment = yield payment.to(path(['transferIn', 'details', 'account'], trade))
       }
 
-      payment = yield payment.description(`Exchange Trade COINIFY=${trade.id}`)
-      payment = yield payment.build()
+      payment = yield payment.description(`Exchange Trade CNY-${trade.id}`)
+      try {
+        payment = yield payment.build()
+      } catch (e) {
+        yield put(actions.logs.logErrorMessage(logLocation, 'sell', e))
+      }
 
-      yield put(sendBtcActions.sendBtcPaymentUpdatedSuccess(payment.value()))
       payment = yield payment.sign(password)
       payment = yield payment.publish()
-      yield put(sendBtcActions.sendBtcPaymentUpdatedSuccess(payment.value()))
+
+      yield put(actions.core.data.bitcoin.fetchData())
+      yield put(actions.core.wallet.setTransactionNote(payment.value().txId, payment.value().description))
+
       yield put(A.coinifySuccess())
       yield put(actions.form.change('buySellTabStatus', 'status', 'order_history'))
       yield put(actions.modals.showModal('CoinifyTradeDetails', { trade }))
+      yield call(initializePayment)
```

Comment:
we must reinitialize through the action IMO, it is not the part of sending funds business logic, and it would also be nice to log it in devtools

Reply:
Good call, I agree.